calicoctl/resourcemgr/hostendpoint.go

```diff
@@ -47,14 +47,15 @@ func init() {
 		},
 		func(ctx context.Context, client client.Interface, resource ResourceObject) (ResourceObject, error) {
 			r := resource.(*api.HostEndpoint)
-			return client.HostEndpoints().Delete(ctx, r.Name, options.DeleteOptions{})
+			return client.HostEndpoints().Delete(ctx, r.Name, options.DeleteOptions{ResourceVersion: r.ResourceVersion})
 		},
 		func(ctx context.Context, client client.Interface, resource ResourceObject) (ResourceObject, error) {
 			r := resource.(*api.HostEndpoint)
-			return client.HostEndpoints().Get(ctx, r.Name, options.GetOptions{})
+			return client.HostEndpoints().Get(ctx, r.Name, options.GetOptions{ResourceVersion: r.ResourceVersion})
 		},
 		func(ctx context.Context, client client.Interface, resource ResourceObject) (ResourceListObject, error) {
-			return client.HostEndpoints().List(ctx, options.ListOptions{})
+			r := resource.(*api.HostEndpoint)
+			return client.HostEndpoints().List(ctx, options.ListOptions{ResourceVersion: r.ResourceVersion, Name: r.Name})
```

Comment:
why do we need a `r.Name` in the `ListOptions`? this function only gets called when the name is not specified.

Reply:
it's for future work with AAPIS - they needed the ability to List/Watch specifying name.

Reply:
incidentally, I think it is still correct for GetOrList to fan out to Get when the resource is fully qualified - because in that case we don't want to display a ResourceList.